nixos/modules/services/web-apps/prosody-filer.nix

```diff
@@ -0,0 +1,63 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+
+  cfg = config.services.prosody-filer;
+
+  settingsFormat = pkgs.formats.toml { };
+  configFile = settingsFormat.generate "prosody-filer.toml" cfg.settings;
+in {
+
+  options = {
+    services.prosody-filer = {
+      enable = mkEnableOption "Prosody Filer XMPP upload file server";
+
+      settings = mkOption {
+        description = "Configuration for Prosody Filer.";
+
+        type = types.submodule {
+          freeformType = settingsFormat.type;
+        };
```

Comment:
is the extra submodule necessary here? `formats.toml` already produces an `attrsOf toml` type, we think it should work without wrapping it in a submodule